Mathlib/MeasureTheory/Function/SimpleFunc.lean

```diff
@@ -553,6 +553,78 @@ instance instModule [Semiring K] [AddCommMonoid β] [Module K β] : Module K (α
 theorem smul_eq_map [SMul K β] (k : K) (f : α →ₛ β) : k • f = f.map (k • ·) :=
   rfl
 
+lemma smul_const [SMul K β] (k : K) (b : β) :
+    (k • const α b : α →ₛ β) = const α (k • b) := ext fun _ ↦ rfl
+
+lemma one_eq_const_one [One β] : (1 : α →ₛ β) = const α 1 := rfl
+
+instance [NonUnitalNonAssocSemiring β] : NonUnitalNonAssocSemiring (α →ₛ β) where
+  left_distrib _ _ _ := ext fun _ ↦ left_distrib ..
+  right_distrib _ _ _ := ext fun _ ↦ right_distrib ..
+  zero_mul _ := ext fun _ ↦ zero_mul ..
+  mul_zero _ := ext fun _ ↦ mul_zero ..
+
+instance [NonAssocSemiring β] : NonAssocSemiring (α →ₛ β) where
+  one_mul _ := ext fun _ ↦ one_mul ..
+  mul_one _ := ext fun _ ↦ mul_one ..
+
+instance [NonUnitalSemiring β] : NonUnitalSemiring (α →ₛ β) where
+  mul_assoc _ _ _ := ext fun _ ↦ mul_assoc ..
+
+instance [Semiring β] : Semiring (α →ₛ β) where
+
+instance [NonUnitalRing β] : NonUnitalRing (α →ₛ β) where
+
+instance [Ring β] : Ring (α →ₛ β) where
+
+instance [Monoid K] [MulAction K β] : MulAction K (α →ₛ β) where
```

Comment:
This would make more sense before the Module instance higher up in the file, as it is weaker

Reply:
Done.